conans/client/shims/write_shims.py

```diff
@@ -0,0 +1,64 @@
+import os
+import stat
+import textwrap
+
+from jinja2 import Template
+
+from conans.client.envvars.environment import BAT_FLAVOR, SH_FLAVOR, env_files
+from conans.client.run_environment import RunEnvironment
+from conans.util.files import save_files
+
+cmd_template = textwrap.dedent("""\
+    @echo off
+    call "{{ activate_path }}"
+    pushd "{{ exe_path_dirname }}"
+    call "{{ exe_path }}" %*
+    popd
+    call "{{ deactivate_path }}"
+    """)
+
+sh_template = textwrap.dedent("""\
+    #!/bin/bash
+    source "{{ activate_path }}"
+    pushd "{{ exe_path_dirname }}" > /dev/null
+    "{{ exe_path }}" "$@"
+    popd > /dev/null
+    source "{{ deactivate_path }}"
+    """)
+
+
+def _generate_shim(name, conanfile, settings_os, output_path):
```

Comment:
Is this a new Generator like the toolchains? something you can parameterize and control too?